src/main/webapp/WEB-INF/pages/productDescription.jsp

```diff
@@ -38,24 +36,45 @@
         </td>
       </tr>
   </table>
-        <form method = "post" >
-              <input type ="cart" name ="${not empty param.quontity ? param.quontity : 1}">
+        <form method="post" >
+              <label>Quantity</label><input type="text" name ="quantity" value="${not empty param.quantity ? param.quantity : 1}">
               <button>Add to cart</button>
-        </form>   
-        <c:if test="${error!=null}">
-            <p><font color="red"><c:out value ="${error}"/></font></p>
-        </c:if>
-        <c:if test="${error==null}">
+        </form>
         <p>
-            <font color="green">
-            <c:out value ="${param.message}"/>
-            </font>
+            <c:choose>
+                <c:when test="${error!=null}">
+                    <span style="color: red; "><c:out value="${error}"/></span>
+                </c:when>
+                <c:otherwise>
+                    <span style="color: green; "><c:out value="${param.message}"/></span>
+                </c:otherwise>
+            </c:choose>
         </p>
-        </c:if>
-        <c:set var="list" value="${sessionScope['cartList'].list}"/>
+
+    <c:set var="list" value="${sessionScope['cartList'].list}"/>
         <c:if test="${list!=null}">
             <c:forEach items="${list}" var="cart" >
                 <p><c:out value="${cart}"/></p>
             </c:forEach>
-        </c:if>      
+        </c:if>
+    <c:set var="list" value="${sessionScope['viewed'].list}"/>
+        <c:if test="${list!=null}">
+            <c:forEach items="${list}" var="view">
+            <p>
```

Comment:
```
<p>
            <table>
                <td>
                   <p>
```

Do we need this? table inside p, p inside td. No <tr> tag

Please check proper table layout: https://www.w3schools.com/html/html_tables.asp

Please watch for IntelliJ IDEA hints (yellow marks on the right, colored strings) - they can find errors.